addon/list-utils.js

```diff
@@ -176,3 +195,19 @@ function normalizeItems ({data, labelAttribute, records, valueAttribute}) {
     })
   )
 }
+
+/**
+ * Determine whether or not valueRecord should be appended to data
+ * @param {Object[]} data - the larger set of data
+ * @param {EmberObject} valueRecord - the record to add or not
+ * @param {String} labelAttribute - dot notated path to label attribute in record
+ * @param {String} valueAttribute - dot notated path to value attribute in record
+ * @param {String} filter - the partial match query filter to populate
+ * @returns {boolean} true if valueRecord should be added to data
+ */
+function shouldAddCurrentValue ({data, valueRecord, labelAttribute, valueAttribute, filter}) {
+  const filterRegex = new RegExp(filter, 'i')
+  return filterRegex.test(valueRecord.get(labelAttribute)) && !data.find(item => {
+    return item.value === valueRecord.get(valueAttribute)
+  })
```

Comment:
I'd probably do something like this so it is more clear to someone else exactly what this is doing:

```js
function shouldAddCurrentValue ({data, filter, labelAttribute, valueAttribute, valueRecord}) {
  const filterRegex = new RegExp(filter, 'i')
  const selectedItemMatchesFilter = filterRegex.test(valueRecord.get(labelAttribute))
  const itemsContainSelectedItem = data.find(item => item.value === valueRecord.get(valueAttribute))
  return selectedItemMatchesFilter && !itemsContainSelectedItem
}
```